Anyone understands why we get the warning `calling host function from device function` when we remove the `constexpr` qualification from the `SharedAllocationTracker` constructor ?
```
/builds/kokkos/kokkos/core/src/View/Kokkos_BasicView.hpp(179): error #20011-D: calling a __host__ function("Kokkos::Impl::ReferenceCountedDataHandle<int,  ::Kokkos::HostSpace> ::ReferenceCountedDataHandle()") from a __host__ __device__ function("Kokkos::Impl::BV::BasicView<int,  ::Kokkos::extents<unsigned long, (unsigned long)18446744073709551615ul > ,  ::Kokkos::Experimental::layout_right_padded<(unsigned long)18446744073709551615ul> ,  ::Kokkos::Impl::SpaceAwareAccessor< ::Kokkos::HostSpace,  ::Kokkos::Impl::ReferenceCountedAccessor<int,  ::Kokkos::HostSpace,  ::Kokkos::default_accessor<int> > > > ::BasicView") is not allowed
Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"
1 error detected in the compilation of "/builds/kokkos/kokkos/build/containers/unit_tests/serial/TestSerial_WithoutInitializing.cpp".
gmake[2]: *** [containers/unit_tests/CMakeFiles/Kokkos_ContainersUnitTest_Serial.dir/build.make:244: containers/unit_tests/CMakeFiles/Kokkos_ContainersUnitTest_Serial.dir/serial/TestSerial_WithoutInitializing.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
```

see the log https://gitlab.spack.io/kokkos/kokkos/-/jobs/17280388

--- a/core/src/impl/Kokkos_SharedAlloc.hpp
+++ b/core/src/impl/Kokkos_SharedAlloc.hpp
@@ -607,8 +607,8 @@ union SharedAllocationTracker {
   ~SharedAllocationTracker(){KOKKOS_IMPL_SHARED_ALLOCATION_TRACKER_DECREMENT}
 
   KOKKOS_FORCEINLINE_FUNCTION
-#if !defined(KOKKOS_COMPILER_GNU) || (KOKKOS_COMPILER_GNU < 1220) || \
-    (KOKKOS_COMPILER_GNU > 1240)
+#if defined(KOKKOS_COMPILER_NVCC) || !defined(KOKKOS_COMPILER_GNU) || \
+    (KOKKOS_COMPILER_GNU < 1220) || (KOKKOS_COMPILER_GNU > 1240)
       // FIXME_GCC: The ViewSupport test fails with gcc 12.2, 12.3 and 12.4
       // because this constructor is optimized out, which leads to a nullptr
       // dereference. Removing the constexpr fixes the issue but nvcc complains,

